core/src/components/input/input.tsx

```diff
@@ -504,9 +504,11 @@ export class Input implements ComponentInterface {
   private getSize() {
     const theme = getIonTheme(this);
     const { size } = this;
-    if (theme !== 'ionic' && size === 'large') {
+    const ionicSizes = ['large', 'xlarge'];
+    if (theme !== 'ionic' && size && ionicSizes.includes(size)) {
```

Comment:
~ nit: Creating an array and checking if a value exists in it is computationally slower than checking the two cases of if the size is large or xlarge. We could probably simplify that so this practice isn't copied into other areas.

Reply:
Fixed: https://github.com/ionic-team/ionic-framework/pull/29446/commits/59e79a3f1e19777d61a62e1955b953fd36a064f4